src/Form/FormBuilder.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Retrofit\Drupal\Form;
+
+use Drupal\Core\Form\FormBuilder as CoreFormBuilder;
+use Drupal\Core\Form\FormStateInterface;
+use Retrofit\Drupal\Form\ArrayAccessFormState;
+
+class FormBuilder extends CoreFormBuilder
+{
+    /**
+     * @return mixed[]
+     */
+    public function getForm(mixed $form_arg): array
+    {
+        $form_state = new ArrayAccessFormState();
+        $args = func_get_args();
+        array_shift($args);
+        $form_state->addBuildInfo('args', $args);
+        return $this->buildForm($form_arg, $form_state);
+    }
```

Comment:
Okay, we are decorating the form builder. Can we override `buildForm` instead?

```php
public function buildForm($form_arg, FormStateInterface &$form_state) 
{
	$form_state = (new ArrayAccessFormState())->setFormState($form_state_additions);
	return parent::buildForm($form_arg, $form_state);
}
```

Reply:
Possibly this:
```
public function buildForm($form_arg, FormStateInterface &$form_state) 
{
	$form_state = (new ArrayAccessFormState())->setFormState($form_state->getCacheableArray());
	return parent::buildForm($form_arg, $form_state);
}
```

Reply:
Doesn't work. If a property doesn't exist the value gets added to the storage property.

Reply:
Okay, so what about the pattern here; 

```
    $original_form_state = $form_state;
    $form_state = new ArrayAccessFormState();
    foreach ($original_form_state as $offset => $value) {
        $form_state[$offset] = $value;
    }
```

Could that be used inside of `buildForm`

Reply:
Something close to that worked.